docs/TemplateManagementCLI.md

```diff
@@ -0,0 +1,100 @@
+Template Management CLI is a tool to manage template files for FHIR Converter DotLiquid engine. This CLI tool manages all template files in [OCI image](https://github.com/opencontainers/image-spec) format and allows pushing and pulling templates with [Azure Container Registry](https://azure.microsoft.com/en-us/services/container-registry/).
+
+Template image is a layer based structure similar to docker image and uses [overlayfs](https://www.kernel.org/doc/html/latest/filesystems/overlayfs.html?highlight=overlayfs) concept to organize templates.
+
+For custom templates, we use two layers image structure to organize template collection: base layer and user layer (The user layer could be extended to multi-layers in the future if necessary). Base layer packs official published templates and user layer packs all modified templates from users. Each layer will be compressed into "*.tar.gz" file before pushing to ACR.
+# Using Template Management CLI
+
+The command-line tool can be used to pull and push a template collection through remote registry (Now we only support Azure Container Registry). 
+
+## Prerequisites
+* Azure container registry - Create a container registry in your Azure subscription. For example, use the [Azure portal](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal) or the [Azure CLI](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-azure-cli).
+
+* Azure Active Directory service principal (optional) - If using service principal's identity for authentication, you need to create a [service principal](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) to access your registry. Ensure that the service principal is assigned a role such as AcrPush so that it has permissions to push and pull artifacts.
+
+* Azure CLI (optional) - To use an individual identity, you need a local installation of the Azure CLI. Version 2.0.71 or later is recommended. Run `az --version` to find the version. If you need to install or upgrade, see [Install Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+
+* Docker (optional) - To use an individual identity, you must also have Docker installed locally, to authenticate with the registry. Docker provides packages that easily configure Docker on any [macOS](https://docs.docker.com/docker-for-mac/), [Windows](https://docs.docker.com/docker-for-windows/), or [Linux](https://docs.docker.com/engine/install/) system.
+## Authentication
+
```

Comment:
Are all the three above optional? Can a user use the tool without having any of the three? Or, or is it that user can use either service principal or individual identity. If it is later, clarify this in the beginning.

Reply:
Since we already pack oras tool in our package, user can use oras tool for authentication. 
I reference the Prerequisites part in this link https://docs.microsoft.com/en-us/azure/container-registry/container-registry-oci-artifacts#sign-in-to-a-registry